testing/jormungandr-testing-utils/src/testing/fragments/sender.rs

```diff
@@ -142,6 +142,21 @@ impl<'a, S: SyncNode + Send> FragmentSender<'a, S> {
         self.send_fragment(from, fragment, via)
     }
 
+    pub fn send_transaction_with_validity<A: FragmentNode + SyncNode + Sized + Send>(
+        &self,
+        from: &mut Wallet,
+        to: &Wallet,
+        via: &A,
+        value: Value,
+        valid_until: BlockDate,
+    ) -> Result<MemPoolCheck, FragmentSenderError> {
+        let address = to.address();
+        let fragment =
+            from.transaction_to(&self.block0_hash, &self.fees, valid_until, address, value)?;
```

Comment:
This is misleading, the wallet treats the date as "current date" (poor choice of name there, besides the unwanted manipulation), so this does not set the expiry date for the transaction to `valid_until`, but to the next epoch.
I think this is another evidence that `Wallet` should not do anything with the date it receives, and use it directly to set the `valid_until` transaction field